soc/arm/microchip_mec/mec172x/device_power.c

```diff
@@ -0,0 +1,384 @@
+#define PCR_XEC_REG_BASE						\
+	((struct pcr_regs *)(DT_REG_ADDR(DT_NODELABEL(pcr))))
+#define TFDP_0_XEC_REG_BASE						\
+	((struct tfdp_regs *)(DT_REG_ADDR(DT_NODELABEL(tfdp0))))
+#define UART_0_XEC_REG_BASE						\
+	((struct uart_regs *)(DT_REG_ADDR(DT_NODELABEL(uart0))))
+#define UART_1_XEC_REG_BASE						\
+	((struct uart_regs *)(DT_REG_ADDR(DT_NODELABEL(uart1))))
+#define VBATR_XEC_REG_BASE						\
+	((struct vbatr_regs *)(DT_REG_ADDR_BY_NAME(DT_NODELABEL(pcr), vbatr)))
+#define VBATM_XEC_BASE_ADDR						\
+	((uintptr_t)(DT_REG_ADDR(DT_NODELABEL(bbram))))
+
+#define BTMR16_0_ADDR	DT_REG_ADDR(DT_NODELABEL(timer0))
+#define BTMR16_1_ADDR	DT_REG_ADDR(DT_NODELABEL(timer1))
+#define BTMR16_2_ADDR	DT_REG_ADDR(DT_NODELABEL(timer2))
+#define BTMR16_3_ADDR	DT_REG_ADDR(DT_NODELABEL(timer3))
+#define BTMR32_0_ADDR	DT_REG_ADDR(DT_NODELABEL(timer4))
+#define BTMR32_1_ADDR	DT_REG_ADDR(DT_NODELABEL(timer5))
+#define VBATM_XEC_ADDR	DT_REG_ADDR(DT_NODELABEL(vbr))
+
+#ifdef DEBUG_DEEP_SLEEP_CLK_REQ
+void soc_debug_sleep_clk_req(void)
+{
+	struct ecs_regs *ecs = ECS_XEC_REG_BASE;
+	struct pcr_regs *pcr = PCR_XEC_REG_BASE;
+	uintptr_t vbm_addr = VBATM_XEC_BASE_ADDR;
+
+	/* Save status to debug LPM been blocked */
+	for (int i = 0; i < 5; i++) {
+		sys_write32(pcr->CLK_REQ[i], vbm_addr);
+		vbm_addr += 4;
+	}
+
+	sys_write32(pcr->SYS_SLP_CTRL, vbm_addr);
+	vbm_addr += 4;
+	sys_write32(ecs->SLP_STS_MIRROR, vbm_addr);
+}
+#endif
+
+/*
+ * Allow peripherals connected to external masters to wake the PLL but not
+ * the EC. Once the peripheral has serviced the external master the PLL
+ * will be turned back off. For example, if the eSPI master requests eSPI
+ * configuration information or state of virtual wires the EC doesn't need
+ * to be involved. The hardware can power on the PLL long enough to service
+ * the request and then turn the PLL back off.  The SMBus and I2C peripherals
+ * in slave mode can also make use of this feature.
+ */
+void soc_deep_sleep_non_wake_en(void)
+{
+#ifdef CONFIG_ESPI_XEC
+	struct ecia_named_regs *regs = ECIA_XEC_REG_BASE;
+
+	regs->GIRQ22.SRC = UINT32_MAX;
+	regs->GIRQ22.EN_SET = MCHP_ESPI_WK_CLK_GIRQ_BIT;
+#endif
+}
+
+void soc_deep_sleep_non_wake_dis(void)
+{
+#ifdef CONFIG_ESPI_XEC
+	struct ecia_named_regs *regs = ECIA_XEC_REG_BASE;
+
+	regs->GIRQ22.EN_CLR = UINT32_MAX;
+	regs->GIRQ22.SRC = UINT32_MAX;
+#endif
+}
+
+/* When MEC172x drivers are power-aware this should be move there */
+void soc_deep_sleep_wake_en(void)
+{
+#if defined(CONFIG_KSCAN) || defined(CONFIG_PS2_XEC_0)
+	struct ecia_named_regs *regs = ECIA_XEC_REG_BASE;
+#if defined(CONFIG_KSCAN)
+	/* Enable PLL wake via KSCAN  */
+	regs->GIRQ21.SRC = MCHP_KEYSCAN_GIRQ_BIT;
+	regs->GIRQ21.EN_SET = MCHP_KEYSCAN_GIRQ_BIT;
+#endif
+#if defined(CONFIG_PS2_XEC_0)
+	/* Enable PS2_0B_WK */
+	regs->GIRQ21.SRC = MCHP_PS2_0_PORT0B_WK_GIRQ_BIT;
+	regs->GIRQ21.EN_SET = MCHP_PS2_0_PORT0B_WK_GIRQ_BIT;
+#endif
+#endif
+}
+
+void soc_deep_sleep_wake_dis(void)
+{
+#ifdef CONFIG_PS2_XEC_0
```

Comment:
replace this with device tree checks as pointed in recent PS2 driver PR

Reply:
updated